processor/src/test/java/com/linecorp/decaton/processor/processors/BatchingProcessorTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.decaton.processor.processors;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.verify;
+import static org.mockito.internal.verification.VerificationModeFactory.times;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import com.linecorp.decaton.processor.Completion;
+import com.linecorp.decaton.processor.ProcessingContext;
+import com.linecorp.decaton.protocol.Sample.HelloTask;
+
+public class BatchingProcessorTest {
+    @Rule
+    public MockitoRule rule = MockitoJUnit.rule();
+
+    @Mock
+    private ProcessingContext<HelloTask> context;
+
+    @Mock
+    private Completion completion;
+
+    private final List<HelloTask> processedTasks = new ArrayList<>();
```

Comment:
Maybe this should be wrapped with `Collections.synchronizedList` ? Because this will be read from test thread while mutated by BatchingProcessor's executor thread.

Reply:
I fixed it. a882452